base/dcim.yml

```diff
@@ -348,178 +467,158 @@ nodes:
-            color: "#faa446"
-          - name: server
-            label: Server
-            description: Dedicated to interfaces connected to servers.
-            color: "#98b2d1"
-          - name: loopback
-            label: Loopback
-            description: Internal interface.
-            color: "#93e9be"
-          - name: management
-            label: Management
-            description: Dedicated network for control and monitoring.
-            color: "#ccc28a"
-          - name: uplink
-            label: Uplink
-            description: Connect to higher device.
-            color: "#ff6b6b"
-          - name: leaf
-            label: Leaf
-            description: Connect to leaf device.
-            color: "#e6e6fa"
-          - name: spare
-            label: Spare
-            description: Reserve resources for future use.
-            color: "#d3d3d3"
+        cardinality: many
+        kind: Attribute
+        identifier: "physcial__breakout"
+        direction: outbound
+        description: "Interfaces resulting from the breakout"
+        order_weight: 1650
+      - name: parent_interface
+        peer: InterfacePhysical
+        optional: true
+        cardinality: one
+        kind: Attribute
+        identifier: "physcial__breakout"
+        direction: inbound
+        description: "Interface from which breakout is created"
         order_weight: 1700
-      - name: status
-        kind: Dropdown
+      - name: children_interfaces
+        label: Sub-interfaces
+        peer: InterfaceSub
         optional: true
-        order_weight: 1300
-        choices:
-          - name: active
-            label: Active
-            description: Fully operational and currently in service.
-            color: "#7fbf7f"
-          - name: provisioning
-            label: Provisioning
-            description: In the process of being set up and configured.
-            color: "#ffff7f"
-          - name: maintenance
-            label: Maintenance
-            description: Undergoing routine maintenance or repairs.
-            color: "#ffd27f"
-          - name: drained
-            label: Drained
-            description: Temporarily taken out of service.
-            color: "#bfbfbf"
+        cardinality: many
+        kind: Attribute
+        identifier: "interface__parent"
+        direction: inbound
+        description: "Sub-interfaces of the Physical interface"
+        order_weight: 1750
+      - name: lacp_parent
+        label: "LAG"
+        peer: InterfaceLag
+        identifier: "lag__members"
+        optional: true
+        cardinality: one
+        kind: Attribute
+        description: "LACP Interface using this Physical interface"
+        order_weight: 1800
+
+  - name: Virtual
+    namespace: Interface
+    label: Virtual Interface
+    description: "Virtual interface like VLAN or Loopback"
+    inherit_from:
+      - DcimInterface
+      - InterfaceLayer2
+      - InterfaceLayer3
```

Comment:
But here your virtual interface could have both l2 and l3 configurations ... wondering if we shouldn't create multiple nodes for that

e.g. 
virtual l2 interface
virtual l3 interface

Reply:
it's a tradeoff, if you get 2 types (L2 and L3) for virtual, you may want to do it for other interfaces too, but if you then want to keep some relationships no too  "wide", you will need to create other Generics, just for the sake of "grouping" some models together.